pymc/tests/distributions/test_multivariate.py

```diff
@@ -1367,6 +1391,100 @@ def test_issue_3706(self):
         assert prior_pred["X"].shape == (1, N, 2)
 
 
+COORDS = {
+    "regions": ["a", "b", "c"],
+    "answers": ["yes", "no", "whatever", "don't understand question"],
+}
+
+
+class TestZeroSumNormal:
+    @pytest.mark.parametrize(
+        "dims, zerosum_axes, shape",
+        [
+            (("regions", "answers"), "answers", None),
+            (("regions", "answers"), ("regions", "answers"), None),
+            (("regions", "answers"), 0, None),
+            (("regions", "answers"), -1, None),
+            (("regions", "answers"), (0, 1), None),
+            (None, -2, (len(COORDS["regions"]), len(COORDS["answers"]))),
+        ],
+    )
+    def test_zsn_dims_shape(self, dims, zerosum_axes, shape):
+        with pm.Model(coords=COORDS) as m:
+            v = pm.ZeroSumNormal("v", dims=dims, shape=shape, zerosum_axes=zerosum_axes)
+            s = pm.sample(10, chains=1, tune=100)
+
+        # to test forward graph
+        random_samples = pm.draw(
+            v,
+            draws=10,
+        )
+
+        assert s.posterior.v.shape == (1, 10, len(COORDS["regions"]), len(COORDS["answers"]))
+
+        if not isinstance(zerosum_axes, (list, tuple)):
+            zerosum_axes = [zerosum_axes]
+
+        if isinstance(zerosum_axes[0], str):
+            for ax in zerosum_axes:
+                for samples in [
+                    s.posterior.v.mean(dim=ax),
+                    random_samples.mean(axis=dims.index(ax) + 1),
+                ]:
+                    assert np.isclose(
+                        samples, 0
+                    ).all(), f"{ax} is a zerosum_axis but is not summing to 0 across all samples."
+
+            nonzero_axes = list(set(dims).difference(zerosum_axes))
+            if nonzero_axes:
+                for ax in nonzero_axes:
+                    for samples in [
+                        s.posterior.v.mean(dim=ax),
+                        random_samples.mean(axis=dims.index(ax) + 1),
+                    ]:
+                        assert not np.isclose(
+                            samples, 0
```

Comment:
You should draw more samples and compare the standard deviation to the desired value imposed by the distribution

Reply:
But the distribution doesn't impose any standard deviation value to ZSN, IIUC: the 1 in `sigma` refers to the underlying, unconstrained Normal distribution

Reply:
Yes, but if you set sigma to 2 or something else, the variance across draws will be determined by the sigma value

Reply:
In the 1d case the marginal variance should be `sigma ** 2 * (n - 1) / n` (from the covariance in the definition)

Reply:
Ah thanks @aseyboldt , I'll test against this formula

Reply:
That works! Added a test 😉